src/audio/eq_iir/eq_iir.c

```diff
@@ -730,6 +764,39 @@ static int eq_iir_get_config(struct processing_module *mod,
 	return comp_data_blob_get_cmd(cd->model_handler, cdata, fragment_size);
 }
 
+static int eq_iir_new_blob(struct processing_module *mod, struct comp_data *cd,
+			   enum sof_ipc_frame source_format, enum sof_ipc_frame sink_format,
+			   int channels)
+{
+	int ret;
+
+	ret = eq_iir_setup(mod, channels);
+	if (ret < 0) {
+		comp_err(mod->dev, "eq_iir_new_blob(), failed IIR setup");
+		return ret;
+	} else if (cd->iir_delay_size) {
+		comp_dbg(mod->dev, "eq_iir_new_blob(), active");
+#if CONFIG_IPC_MAJOR_3
+		cd->eq_iir_func = eq_iir_find_func(source_format, sink_format, fm_configured,
+						   ARRAY_SIZE(fm_configured));
+#endif
+#if CONFIG_IPC_MAJOR_4
```

Comment:
using `#elif` helps show that IPC3 and IPC4 are mutually exclusive. Elsewhere too of course

Reply:
Yep, fix coming